exchanges/auth/src/authExchange.ts

```diff
@@ -103,28 +103,32 @@ export function authExchange<T>({
 
       const updateAuthState = (newAuthState: T | null) => {
         authState = newAuthState;
-        pendingPromise = undefined;
+        authPromise = undefined;
         retryQueue.forEach(retryOperation);
         retryQueue.clear();
       };
 
-      let pendingPromise: Promise<any> | void = Promise.resolve()
+      let authPromise: Promise<any> | void = Promise.resolve()
         .then(() => getAuth({ authState, mutate }))
         .then(updateAuthState);
 
-      const refreshAuth = (operation: Operation): Promise<any> => {
+      const refreshAuth = (
+        operation: Operation,
+        addAuthAttempt = true
```

Comment:
We might want to invert this to `skipAddAuthAttempt` this because the polyfill used for `argument defaulting` is extremely expensive and will cause a big bump in the bundle size of this exchange 😅

Reply:
reviewed